src/modules/site-v2/base/static/js/scatter-histogram.js

```diff
@@ -0,0 +1,253 @@
+
+    // Read values from config, filling in default values when not supplied
+    const bins_per_tick = config['bins_per_tick'] || 1;
+    const circle_radius = config['circle_radius'] || 4;
+    const fill_color    = config['fill_color']    || 'black';
+    const stroke_color  = config['stroke_color']  || 'none';
+
+    // Create a template function for the tooltip
+    // By default, show label & value for each axis
+    const tooltip_template = config['tooltip_template'] || ((d, labels) => `
+      <p>${labels[0]}: ${d[0]}</p>
+      <p>${labels[1]}: ${d[1]}</p>
+    `);
+
+    // Create the SVG object for the full graphic (scatterplot + histograms + margins)
+    const svg = d3.select(container_selector).append('svg')
+      .attr('width',  width  + margin.left + margin.right + hist_height)
+      .attr('height', height + margin.top + margin.bottom + hist_height)
+
+    // Add a graph element for the scatterplot
+    const g = svg.append("g")
+      .attr("transform", `translate(${ margin.left }, ${ margin.top + hist_height })`);
+
+    // Create mapping functions from data set to coordinates in the scatterplot
+    // Add a buffer of 5 in either direction so that dots aren't directly on graph edge
+    const x = d3.scaleLinear().domain(buffered_extent(data, 5, n => n[0])).range([0, width]).nice();
+    const y = d3.scaleLinear().domain(buffered_extent(data, 5, n => n[1])).range([height, 0]).nice();
+
+    // Add the x-axis
+    g.append("g")
+      .attr("transform", `translate(0, ${ height })`)
+      .call(d3.axisBottom(x));
+
+    // Add the y-axis
+    g.append("g")
+      .call(d3.axisLeft(y));
+
+    // Add label for x-axis, if one is provided
+    // Centered on scatterplot graph element
+    if (config['x_label']) {
+      svg.append('text')
+        .attr('x', margin.left + (width / 2))
+        .attr('y', height + margin.top + hist_height + 36)
+        .attr('text-anchor', 'middle')
+        .text(config['x_label'])
+    }
+
+    // Add label for y-axis, if one is provided
+    // Centered on scatterplot graph element
+    if (config['y_label']) {
+      svg.append('text')
+        .attr('transform', 'rotate(-90)')
+        .attr('x', -(margin.top + hist_height + (height / 2)))
+        .attr('y', 0)
+        .attr('dy', '.75em')
+        .attr('text-anchor', 'middle')
+        .text(config['y_label'])
+
+      }
+
+    // Add the data points
+    const dots = g.selectAll(".point")
+      .data(data)
+      .enter()
+      .append("circle")
+      .attr("cx", d => x(d[0]) )
+      .attr("cy", d => y(d[1]) )
+      .attr("r", circle_radius)
+      .style('fill',   fill_color)
+      .style('stroke', stroke_color);
+
+    // Create tooltip for data point mouseover
+    const tooltip = d3.select("#phenotype-chart")
+      .append("div")
+      .attr("class", "tooltip")
+      .style("opacity", 0);
+
+    // Add mouseover listener for individual dots
+    dots.on('mouseover', function(d) {
```

Comment:
hum....on mobile we have `touchstart|touchmove|touchend` -- is there parity of experience we can have on mobile/touch devices with the charts?

Reply:
Good question -- moving to Basecamp